controllers/machine_controller.go

```diff
@@ -242,6 +255,73 @@ func (r *MachineReconciler) isDeleteNodeAllowed(ctx context.Context, machine *cl
 	}
 }
 
+func (r *MachineReconciler) drainNode(ctx context.Context, cluster *clusterv1.Cluster, name string) error {
+	var kubeClient kubernetes.Interface
+	if cluster == nil {
+		var err error
+		kubeClient, err = kubernetes.NewForConfig(r.config)
+		if err != nil {
+			return fmt.Errorf("unable to build kube client: %v", err)
```

Comment:
```suggestion
			return errors.Errorf("unable to build kube client: %v", err)
```

Reply:
Do we want to record an event here against the Machine so the user is informed?

Reply:
We don't record an event for this in deleteNode; We should probably record the event there, since we'll fall-through to that method anyway.